user-auth-api.py

```
@@ -0,0 +1,108 @@
+# user_auth_api.py
+# Test file for CodeRabbit security review capabilities
+
+import mysql.connector
+import hashlib
+import pickle
+import os
+from flask import Flask, request, jsonify, session
+
+app = Flask(__name__)
+app.secret_key = "my-secret-key-123"  # Hardcoded secret
```

Ruff (0.12.2)

Possible hardcoded password assigned to: "secret_key"

(S105)